highgui/window_QT.cpp erroneously includes glx.h on GLES plattforms

<!--
If you have a question rather than reporting a bug please go to http://answers.opencv.org where you get much faster responses.
If you need further assistance please read [How To Contribute](https://github.com/opencv/opencv/wiki/How_to_contribute).

This is a template helping you to create an issue which can be processed as quickly as possible. This is the bug reporting section for the OpenCV library.
-->

##### System information (version)
<!-- Example
- OpenCV => 3.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2015
-->

- OpenCV => 3.2
- Operating System / Platform => Linux
- Compiler => GCC7

##### Detailed description

After
https://github.com/opencv/opencv/commit/3bf16da92d1cb42d429e9a81a3a0abd5cf3360a7, which
adresses #6471,

GL/glx.h is included on all LINUX plattforms, which is wrong for a number of reasons:

- GL_PERSPECTIVE_CORRECTION_HINT is defined in GL/gl.h, not glx.h, the latter just includes the former
- GL/gl.h is a Desktop GL header, and should not be included on GLES plattforms
- GL/gl.h is already included via QtOpenGL -> QtGui/qopengl.h on desktop plattforms

Including both GLES and GL headers leads to compilation errors on 32bit GLES plattforms, as it causes
conflicting type declarations:

```
[  729s] In file included from /usr/include/GL/gl.h:2060:0,
[  729s]                  from /usr/include/GL/glx.h:32,
[  729s]                  from /home/abuild/rpmbuild/BUILD/opencv-3.2.0/modules/highgui/src/window_QT.cpp:59:
[  729s] /usr/include/GL/glext.h:468:19: error: conflicting declaration 'typedef ptrdiff_t GLsizeiptr'
[  729s]  typedef ptrdiff_t GLsizeiptr;
[  729s]                    ^~~~~~~~~~
[  729s] In file included from /usr/include/qt5/QtGui/qopengl.h:107:0,
[  729s]                  from /usr/include/qt5/QtGui/QtGui:43,
[  729s]                  from /usr/include/qt5/QtOpenGL/QtOpenGLDepends:4,
[  729s]                  from /usr/include/qt5/QtOpenGL/QtOpenGL:3,
[  729s]                  from /home/abuild/rpmbuild/BUILD/opencv-3.2.0/modules/highgui/src/window_QT.h:46,
[  729s]                  from /home/abuild/rpmbuild/BUILD/opencv-3.2.0/modules/highgui/src/window_QT.cpp:47:
[  729s] /usr/include/GLES3/gl3.h:73:25: note: previous declaration as 'typedef khronos_ssize_t GLsizeiptr'
[  729s]  typedef khronos_ssize_t GLsizeiptr;
```

GL_PERSPECTIVE_CORRECTION_HINT does not exist in GLES 2.0/3.x, and has been deprecated in OpenGL 3.0 core profiles.


##### Steps to reproduce

<!-- to add code example fence it with triple backticks and optional file extension
    ```.cpp
    // C++ code example
    ```
 or attach as .txt or .zip file
-->

--- a/modules/highgui/src/window_QT.cpp
+++ b/modules/highgui/src/window_QT.cpp
@@ -54,9 +54,12 @@
 #include <unistd.h>
 #endif
 
+// Get GL_PERSPECTIVE_CORRECTION_HINT definition, not available in GLES 2 or
+// OpenGL 3 core profile or later
 #ifdef HAVE_QT_OPENGL
-    #if defined Q_WS_X11 /* Qt4 */ || defined Q_OS_LINUX /* Qt5 */
-        #include <GL/glx.h>
+    #if defined Q_WS_X11 /* Qt4 */ || \
+        (!defined(QT_OPENGL_ES_2) && defined Q_OS_LINUX) /* Qt5 with desktop OpenGL */
+        #include <GL/gl.h>
     #endif
 #endif
 

--- a/modules/highgui/src/window_QT.cpp
+++ b/modules/highgui/src/window_QT.cpp
@@ -3229,7 +3229,9 @@ void OpenGlViewPort::updateGl()
 
 void OpenGlViewPort::initializeGL()
 {
+#ifdef GL_PERSPECTIVE_CORRECTION_HINT
     glHint(GL_PERSPECTIVE_CORRECTION_HINT, GL_NICEST);
+#endif
 }
 
 void OpenGlViewPort::resizeGL(int w, int h)

